src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -164,21 +164,34 @@ partitions_request_iterator validate_partitions(
       begin,
       valid_partitions_end,
       std::back_inserter(invalid_partitions),
-      error_code::invalid_replication_factor,
+      error_code::unknown_topic_or_partition,
```

Comment:
The error string does not match the error message

Reply:
I thought we discussed this in sync yesterday.
https://github.com/redpanda-data/redpanda/pull/8849#discussion_r1113724399

TLDR - return a retriable error on invalid replication factor check. The error `invalid_replication_factor` is not retriable.